src/System.Private.CoreLib/src/System/Environment.CoreRT.Unix.cs

```diff
@@ -13,23 +14,69 @@ namespace System
     public static partial class Environment
     {
         internal static int CurrentNativeThreadId => ManagedThreadId.Current;
+
+        private static readonly IDictionary<string, string> _environment = GetSystemEnvironmentVariables();
```

Comment:
It would be a bit more efficient as `Dictionary`, not need to pay the interface overhead for the calls here.